test/ibmq/test_ibmq_job_attributes.py

```diff
@@ -207,18 +208,33 @@ def _mocked__api_job_submit(*args, **kwargs):
         self.assertEqual(job.batman, 'bruce')
 
     @requires_provider
-    def test_queue_position(self, provider):
-        """Test retrieving queue position."""
+    def test_queue_info(self, provider):
+        """Test retrieving queue information."""
         # Find the most busy backend.
-        backend = max([b for b in provider.backends() if b.status().operational],
+        backend = max([b for b in provider.backends(simulator=False) if b.status().operational],
                       key=lambda b: b.status().pending_jobs)
         qobj = assemble(transpile(self._qc, backend=backend), backend=backend)
+        leave_states = list(JOB_FINAL_STATES) + [JobStatus.RUNNING]
         job = backend.run(qobj)
-        status = job.status()
-        if status is JobStatus.QUEUED:
-            self.assertIsNotNone(job.queue_position())
+        queue_info = None
+        for _ in range(10):
+            queue_info = job.queue_info()
+            # Even if job status is QUEUED, queue information may not be
+            # immediately available.
+            if (job._status is JobStatus.QUEUED and job.queue_position()) or \
+                    job._status in leave_states:
+                break
+            time.sleep(0.5)
+
+        if job._status is JobStatus.QUEUED:
+            msg = "Job {} is queued but has no ".format(job.job_id())
+            self.assertIsNotNone(job.queue_position(), msg + "queue position.")
```

Comment:
It's explicitly checked because it's a public method, and I don't want to rely on it always being inside `queue_info`. 

I first thought you were asking about lines `224`, which also checks `queue_position()`. This is indeed a duplicate check, but I left it to ensure `queue_position()` is called if `224` changes (since there are multiple ways to do 224).